src/osd/OSD.cc

```diff
@@ -8864,7 +8987,11 @@ void OSD::enqueue_op(PG *pg, OpRequestRef& op)
 	   << " cost " << op->get_req()->get_cost()
 	   << " latency " << latency
 	   << " " << *(op->get_req()) << dendl;
-  pg->queue_op(op);
+  if (pg->can_handle_fast(op)) {
+    pg->queue_op_without_pg_lock(op);
```

Comment:
I misunderstood.  See my other comments.  Don't call it can_handle_fast though, it's too easy to confuse it with the messenger fast_dispatch mechanisms.